tracer/src/Datadog.Trace/Ci/Agent/CIAgentWriter.cs

```diff
@@ -6,22 +6,55 @@
 using System;
 using System.Threading.Tasks;
 using Datadog.Trace.Agent;
-using Datadog.Trace.Ci.Tags;
+using Datadog.Trace.Ci.EventModel;
 using Datadog.Trace.Configuration;
 using Datadog.Trace.Sampling;
 
 namespace Datadog.Trace.Ci.Agent
 {
-    internal class CIAgentWriter : IAgentWriter
+    /// <summary>
+    /// CI Visibility Agent Writer
+    /// </summary>
+    internal class CIAgentWriter : IEventWriter
     {
-        private readonly AgentWriter _agentWriter = null;
-        private readonly bool _isPartialFlushEnabled = false;
+        [ThreadStatic]
+        private static Span[] _spanArray;
+        private readonly AgentWriter _agentWriter;
 
-        public CIAgentWriter(ImmutableTracerSettings settings, ISampler sampler)
+        public CIAgentWriter(ImmutableTracerSettings settings, ISampler sampler, int maxBufferSize = 1024 * 1024 * 10)
         {
-            _isPartialFlushEnabled = settings.Exporter.PartialFlushEnabled;
-            var api = new Api(settings.Exporter.AgentUri, TracesTransportStrategy.Get(settings.Exporter), null, rates => sampler.SetDefaultSampleRates(rates), _isPartialFlushEnabled);
-            _agentWriter = new AgentWriter(api, null, maxBufferSize: settings.TraceBufferSize);
+            var isPartialFlushEnabled = settings.Exporter.PartialFlushEnabled;
+            var apiRequestFactory = TracesTransportStrategy.Get(settings.Exporter);
+            var api = new Api(settings.Exporter.AgentUri, apiRequestFactory, null, rates => sampler.SetDefaultSampleRates(rates), isPartialFlushEnabled);
+            _agentWriter = new AgentWriter(api, null, maxBufferSize: maxBufferSize);
+        }
+
+        public CIAgentWriter(IApi api, ImmutableTracerSettings settings, ISampler sampler, int maxBufferSize = 1024 * 1024 * 10)
```

Comment:
I think you can remove this `ImmutableTracerSettings` parameter

```suggestion
        public CIAgentWriter(IApi api, ISampler sampler, int maxBufferSize = 1024 * 1024 * 10)
```

Reply:
done in [fa1c15a](https://github.com/DataDog/dd-trace-dotnet/pull/2402/commits/fa1c15aea246fa06191980e7d65ee3a8395a59e2)